src/client.ts

```diff
@@ -1,145 +1,215 @@
-import http from 'http';
+import axios, { AxiosInstance } from 'axios';
 
 import { IntegrationProviderAuthenticationError } from '@jupiterone/integration-sdk-core';
 
 import { IntegrationConfig } from './types';
 
 export type ResourceIteratee<T> = (each: T) => Promise<void> | void;
 
-// Providers often supply types with their API libraries.
-
-type AcmeUser = {
-  id: string;
-  name: string;
+type CobaltFinding = {
+  resource: {
+    id: string;
+    tag: string;
+    title: string;
+    description: string;
+    type_category: string;
+    labels: object[];
+    impact: number;
+    likelihood: number;
+    state: string;
+    affected_targets: string[];
+    proof_of_concept: string;
+    suggested_fix: string;
+    prerequisites: string;
+    pentest_id: string;
+    asset_id?: string;
+    log?: object[];
+  };
 };
 
-type AcmeGroup = {
-  id: string;
-  name: string;
-  users?: Pick<AcmeUser, 'id'>[];
+type CobaltAsset = {
+  resource: {
+    id: string;
+    title: string;
+    description: string;
+    asset_type: string;
+    attachments?: object[];
+  };
 };
 
-// Those can be useful to a degree, but often they're just full of optional
-// values. Understanding the response data may be more reliably accomplished by
-// reviewing the API response recordings produced by testing the wrapper client
-// (below). However, when there are no types provided, it is necessary to define
-// opaque types for each resource, to communicate the records that are expected
-// to come from an endpoint and are provided to iterating functions.
-
-/*
-import { Opaque } from 'type-fest';
-export type AcmeUser = Opaque<any, 'AcmeUser'>;
-export type AcmeGroup = Opaque<any, 'AcmeGroup'>;
-*/
+type CobaltPentest = {
+  resource: {
+    id: string;
+    title: string;
+    objectives: string;
+    state: string;
+    tag: string;
+    asset_id: string;
+    platform_tags: string[];
+    methodology: string;
+    targets: string[];
+    start_date: string;
+    end_date: string;
+  };
+};
 
 /**
  * An APIClient maintains authentication state and provides an interface to
  * third party data APIs.
- *
- * It is recommended that integrations wrap provider data APIs to provide a
- * place to handle error responses and implement common patterns for iterating
- * resources.
  */
 export class APIClient {
-  constructor(readonly config: IntegrationConfig) {}
+  #orgToken: string;
+  constructor(readonly config: IntegrationConfig) {
+    this.#orgToken = '';
+  }
+
+  getOrgToken(): string {
+    return this.#orgToken;
+  }
+
+  setOrgToken(orgToken: string) {
+    this.#orgToken = orgToken;
+  }
```

Comment:
```suggestion
  get orgToken(): string {
    return this._orgToken;
  }

  set orgToken(orgToken: string) {
    // ... validate ...
    this._orgToken = orgToken;
  }
```

Allows for `client.orgToken = 'somevalue that will be validated'`. Apart from some kind of processing when setting a value, setters are not typically used in the JS code I've seen.

Reply:
That makes sense. This was some convolution caused by the construction process. I've cleaned it up.